src/SharpLearning.Backend.TensorFlow.Test/TensorFlowRawMnistDeepTest.cs

```diff
@@ -119,32 +148,87 @@ public void MnistDeep()
                     var (testImages, testLabels) = testReader.All();
-
                     TFTensor evaluatedAccuracy = session.GetRunner()
                         .AddInput(x, testImages)
                         .AddInput(y_, testLabels)
-                        .AddInput(keep_prob, test_dropout_keep_prob) // DOESN*T WORK!??!
+                        .AddInput(keep_prob, test_dropout_keep_prob)
                         .Run(accuracy);
-
                     float acc = (float)evaluatedAccuracy.GetValue();
+#else
+                    // Batch evaluation is faster (2x) and uses A LOT less memory, not batched == 6GB, batched == 240MB!
+                    var testBatchSize = 100;
+                    Debug.Assert(data.TestTargets.Data.Length % testBatchSize == 0); // Or we need to do something that handles remaining samples
+                    var rawTestBatchEnumerator = data.CreateTestBatchEnumerator(testBatchSize);
+                    var testBatchEnumerator = Convert(rawTestBatchEnumerator, ClassCount);
+                    var totalCount = 0;
+                    var correctCount = 0;
+                    while (testBatchEnumerator.MoveNext())
+                    {
+                        var (testInputBatch, testLabelBatch) = testBatchEnumerator.CurrentBatch();
+
+                        TFTensor testInputBatchTensor = TFTensor.FromBuffer(new TFShape(testBatchSize, FeatureCount), testInputBatch, 0, testInputBatch.Length);
+                        TFTensor testLabelBatchTensor = TFTensor.FromBuffer(new TFShape(testBatchSize, ClassCount), testLabelBatch, 0, testLabelBatch.Length);
+
+                        TFTensor evaluatedCorrectPrediction = session.GetRunner()
+                            .AddInput(x, testInputBatchTensor)
+                            .AddInput(y_, testLabelBatchTensor)
+                            .AddInput(keep_prob, test_dropout_keep_prob)
+                            .Run(correctPrediction);
 
-                    Log($"Accuracy {acc}");
-                    Assert.AreEqual(0.2029999941587448, acc);
+                        var ps = (bool[])evaluatedCorrectPrediction.GetValue();
+
+                        totalCount += ps.Length;
+                        foreach (var p in ps)
+                        {
+                            if (p) { ++correctCount; }
+                        }
+                    }
+
+                    var incorrectCount = totalCount - correctCount;
+                    var acc = correctCount / (float)totalCount;
+#endif
+                    var testTime_ms = s.Elapsed.TotalMilliseconds;
+                    s.Restart();
+
+                    Log($"Accuracy {acc} Initialize {initializeTime_ms,6:F1} Train {trainTime_ms,6:F1} Test {testTime_ms,6:F1} ");
+                    Assert.AreEqual(0.0979999974370003, acc, 0.00000001); // This is what C# currently computes
+                    Assert.AreEqual(0.2461999952793121, acc); // This is what equivalent python computes
```

Comment:
@mdabros you can see here what the difference is. Not sure why yet. Maybe parameter initialization...